agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/sampling/AiOverrideSampler.java

```diff
@@ -48,7 +57,18 @@ public SamplingResult shouldSample(
       Attributes attributes,
       List<LinkData> parentLinks) {
 
-    Sampler override = samplingOverrides.getOverride(spanKind, attributes);
+    SpanContext parentSpanContext = Span.fromContext(parentContext).getSpanContext();
+
+    // TODO isRequest won't be correct for spring-scheduling, quartz and customInstrumentation
+    //  since we don't have scope available in the sampler (yet)
+    boolean isRequest =
+        SpanDataMapper.isRequest(spanKind, parentSpanContext, null, attributes::get);
+
+    SamplingOverrides samplingOverrides =
+        isRequest ? requestSamplingOverrides : dependencySamplingOverrides;
+
+    boolean standaloneTelemetry = !isRequest && !parentSpanContext.isValid();
+    Sampler override = samplingOverrides.getOverride(standaloneTelemetry, attributes);
```

Comment:
```suggestion
    boolean isStandaloneTelemetry = !isRequest && !parentSpanContext.isValid();
    Sampler override = samplingOverrides.getOverride(isStandaloneTelemetry , attributes);
```

Reply:
conventionally in Java boolean fields are not prefixed with `is`, while their getters are

Reply:
I pushed this change to the next PR to avoid conflicts